include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -93,7 +101,7 @@ class IExecutionProvider {
      3. onnxruntime (framework/session) does not depend on any specific
      execution provider lib.
   */
-  virtual std::shared_ptr<KernelRegistry> GetKernelRegistry() const;
+  virtual std::shared_ptr<KernelRegistry> GetKernelRegistry() const { return nullptr; }
```

Comment:
>{ return nullptr; } [](start = 68, length = 19)

Curious, why this is needed?

Reply:
A opted to move this here for a couple reasons.. one is that trivial default virtual methods might as well be in the header since that's where people look. And the second reason is that with the shared providers putting it here means I don't need to route this function across the DLL boundary as it's just inlined internally.